frontend/src/providers/QueryBuilder.tsx

```diff
@@ -249,6 +251,80 @@ export function QueryBuilderProvider({
 		[getElementWithActualOperator],
 	);
 
+	const isDefaultQuery = useCallback(
+		({ currentQuery, sourcePage }: IsDefaultQueryProps): boolean => {
+			// Get default query with updated operators
+			const defaultQuery = updateAllQueriesOperators(
+				initialQueriesMap[sourcePage],
+				PANEL_TYPES.LIST,
+				sourcePage,
+			);
+
+			// Early return if query types don't match
+			if (currentQuery.queryType !== defaultQuery.queryType) {
+				return false;
+			}
+
+			// Only compare builder queries
+			if (currentQuery.queryType !== EQueryType.QUERY_BUILDER) {
+				return false;
+			}
+
+			const extractRelevantKeys = (queryData: IBuilderQuery): IBuilderQuery => {
+				const {
+					dataSource,
+					queryName,
+					aggregateOperator,
+					aggregateAttribute,
+					timeAggregation,
+					spaceAggregation,
+					functions,
+					filters,
+					expression,
+					disabled,
+					stepInterval,
+					having,
+					groupBy,
+					legend,
+				} = queryData;
+
+				return {
+					dataSource,
+					queryName,
+					aggregateOperator,
+					// remove id from aggregateAttribute
+					aggregateAttribute: {
+						...aggregateAttribute,
+						id: '',
+					},
+					timeAggregation,
+					spaceAggregation,
+					functions,
+					filters,
+					expression,
+					disabled,
+					stepInterval,
+					having,
+					groupBy,
+					legend,
+					// set to default values
+					orderBy: [],
+					limit: null,
+					reduceTo: 'avg',
+				};
+			};
+
+			const currentBuilderData = extractRelevantKeys(
+				currentQuery.builder.queryData[0],
+			);
+			const defaultBuilderData = extractRelevantKeys(
+				defaultQuery.builder.queryData[0],
+			);
+
+			return isEqual(currentBuilderData, defaultBuilderData);
```

Comment:
why are we just comparing the `0` index ? if the current query had multiple queries but the `0`th query actually matched the default query case ?